### PR TITLE
Added importable, exportable, and transformable formats.

### PR DESCRIPTION
**Pull request by David-Rod**
Co-authored-by: Oddant1 als872@nau.edu

This PR adds the importable, exportable, and transformable formats to plugins as properties.

### CODE REVIEW

**Comment by David-Rod** [comment]
Commit https://github.com/qiime2/qiime2/pull/498/commits/b5843b60f3f6bc13090f4b42f90f4e0636dac391 Breaks two tests that reference importable formats. Aside from the test failures, I wanted to commit the changes for feedback. Thanks team.

File: qiime2/plugin/plugin.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from qiime2.plugin.model.base import FormatBase
 from qiime2.core.type import is_semantic_type
 from qiime2.core.util import get_view_name
+import qiime2.core
```
Comment (thermokarst) [inline qiime2/plugin/plugin.py]: Please remove this. :bow_and_arrow:

File: qiime2/plugin/plugin.py [hunk 2]
```diff
@@ -83,6 +84,37 @@ def actions(self):
         actions.update(self.pipelines)
         return types.MappingProxyType(actions)
 
+    @property
+    def importable_formats(self):
```
Comment (thermokarst) [inline qiime2/plugin/plugin.py]: Please remove this. :baby_chick:

File: qiime2/plugin/plugin.py [hunk 3]
```diff
@@ -83,6 +84,37 @@ def actions(self):
...
+            for type_format in self.type_formats:
+                to_type = qiime2.core.transform.ModelType.from_view_type(
+                    type_format.format)
+                if from_type.has_transformation(to_type):
+                    importable_formats[name] = record
+                    break
+        return importable_formats
+
+    @property
+    def exportable_formats(self):
```
Comment (thermokarst) [inline qiime2/plugin/plugin.py]: Please remove this. :fried_egg:

File: qiime2/plugin/plugin.py [hunk 4]
```diff
@@ -83,6 +84,37 @@ def actions(self):
...
+                type_format.format)
+            for name, record in self.formats.items():
+                to_type = qiime2.core.transform.ModelType.from_view_type(
+                    record.format)
+                if from_type.has_transformation(to_type):
+                    exportable_formats[name] = record
+        return exportable_formats
+
+    @property
+    def transformable_formats(self):
```
Comment (thermokarst) [inline qiime2/plugin/plugin.py]: Please remove this. :plate_with_cutlery:

File: qiime2/sdk/plugin_manager.py [hunk 5]
```diff
@@ -105,25 +104,49 @@ def _integrate_plugin(self, plugin):
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
-    @property
-    def importable_formats(self):
-        """Return formats that are importable.
+    def get_formats(self, *, include_all=False, plugin=None):
+        # Initialize formats dictionary
```
Comment (thermokarst) [inline qiime2/sdk/plugin_manager.py]: This is certainly a subjective request, but I don't think that this, or any of the other comments in this PR add value to the source code, they all basically just say the same thing as the code itself. Please remove or revise. :light_rail:

File: qiime2/sdk/plugin_manager.py [hunk 6]
```diff
@@ -105,25 +104,49 @@ def _integrate_plugin(self, plugin):
...
-                to_type = transform.ModelType.from_view_type(
+        # Get all formats if include_all is True
+        if include_all is True:
+            for plugin in plugins.values():
+                formats.update(plugin.formats)
+            return formats
+
+        # Loop through the plugins that we want to get the formats for those
+        # plugins
+        for plugin in plugins:
```
Comment (thermokarst) [inline qiime2/sdk/plugin_manager.py]: I think the logic here can be simplified a bit --- this method should return all the formats that are implicated in a transformation. Let's chat about this IRL when you get a chance.

File: qiime2/plugin/plugin.py (hunk 1 above)
Comment (David-Rod) [reply qiime2/plugin/plugin.py]: Removed

**Comment by thermokarst** [comment]
Hey @David-Rod - this PR looks good to go, but needs a few minor changes prior to merging:

- The test suite [is failing](https://travis-ci.org/qiime2/qiime2/builds/604574412?utm_source=github_status&utm_medium=notification). Please run `make test` locally, you should see the same errors as reported via travis, then, work on updating the broken tests to work as expected. Make sense?
- Fix the merge conflicts that are now present in this PR (I merged @Oddant1's types PR).

Thanks!

**Comment by thermokarst** [comment]
@David-Rod, ready for merge?

**Pull request merged**